roofit/roofitcore/src/RooFormula.cxx

```diff
@@ -512,21 +527,19 @@ void RooFormula::installFormulaOrThrow(const std::string& formula) {
     throw std::runtime_error(msg.str());
   }
 
-  if (theFormula && theFormula->GetNdim() != 1) {
-    // TFormula thinks that we have a multi-dimensional formula, e.g. with variables x,y,z,t.
-    // We have to check now that this is not the case, as RooFit only uses the syntax x[0], x[1], x[2], ...
-    bool haveProblem = false;
-    std::stringstream msg;
-    msg << "TFormula interprets the formula " << formula << " as " << theFormula->GetNdim() << "-dimensional with the variable(s) {";
-    for (int i=1; i < theFormula->GetNdim(); ++i) {
-      const TString varName = theFormula->GetVarName(i);
-      if (varName.BeginsWith("x[") && varName[varName.Length()-1] == ']')
-        continue;
-
-      haveProblem = true;
-      msg << theFormula->GetVarName(i) << ",";
-    }
-    if (haveProblem) {
+  if (theFormula && theFormula->GetNdim() != 0) {
+    auto theNullFormula = std::make_unique<TFormula>((std::string(GetName()) + "0").c_str(), reconstructNullFormula(processedFormula).c_str());
+    if (theNullFormula && theNullFormula->GetNdim() != 0) {
```

Comment:
```suggestion
    TFormula nullFormula{"nullFormula", reconstructNullFormula(processedFormula, /*addToGlobList=*/false).c_str()};
    if (nullFormula.GetNdim() != 0) {
```
1. We generally don't use this `the*` prefix for local variables anymore
2. Can be on the stack
3. `nullptr` check redundant because `make_unique` always constructs something (or throws an exception)
4. Better set `addToGlobList=false` to avoid interacting with `gROOT` for this local object
5. Following from 3, the name now doesn't matter and you can avoid some string manipulation

Reply:
Thanks for the review :)

Question:
1) I copy-pasted from theFormula. Do you want me to replace also theFormula with formula, or just use nullFormula and leave the previous variable untouched?

Reply:
Personally I would keep `theFormula`, just for the sake of unique Ctrl+F-ability in the file :slightly_smiling_face:  `nullFormula` is already unique, so extending it to `theNullFormula` doesn't help.